### ISSUE TITLE
Failure to Launch Forge 36.1.66 or higher

### ISSUE THREAD

**Issue by vPubneutral** [MultiMC/MultiMC5#3953]
### Operating System

Windows

### Description of bug

Attempting to launch Forge 36.1.66 or newer (36.2.0, the next update, is the newest at time of writing) results in an error and failure to launch. The official launcher runs these versions just fine. I have seen other issues saying running in online mode can fix errors like this, but I am running in online mode so I don't think that's the issue.

### Error

Some extra libraries are missing! Run the installer to generate them now.
Failed to start Minecraft:
java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.multimc.onesix.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:210)
	at org.multimc.onesix.OneSixLauncher.launch(OneSixLauncher.java:245)
	at org.multimc.EntryPoint.listen(EntryPoint.java:143)
	at org.multimc.EntryPoint.main(EntryPoint.java:34)
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at io.github.zekerzhayard.forgewrapper.installer.Main.main(Main.java:44)
	... 8 more
Caused by: java.lang.NoSuchMethodError: net.minecraftforge.installer.json.Util.loadInstallProfile()Lnet/minecraftforge/installer/json/Install;
	at io.github.zekerzhayard.forgewrapper.installer.Installer.install(Installer.java:12)
	... 13 more
Exiting with -1
Process exited with code -1.

### Full Log

[https://paste.ee/p/aUagG](https://paste.ee/p/aUagG)

### Steps to reproduce

- Create a 1.16.5 instance
- Add Forge 36.1.66 or newer
- Launch
- Voila

### Suspected cause

Maybe a breaking change on Forge's end. The earliest version that causes this has this for the changelog:

"ForgeGradle 5.1 and Gradle 7, which allows development on J16. Runtime still targets J8." - From [https://maven.minecraftforge.net/net/minecraftforge/forge/1.16.5-36.1.66/forge-1.16.5-36.1.66-changelog.txt](https://maven.minecraftforge.net/net/minecraftforge/forge/1.16.5-36.1.66/forge-1.16.5-36.1.66-changelog.txt)

### This issue is unique

- [x] I have searched the issue tracker and did not find an issue describing my bug.

**Comment by papycochon** [comment]
I think it's because they force the use of minecraft (official launcher) game profiles (at least that's what the error looks like to me)
happens here aswell.

Also what's your Java runtime?
Mine's AdoptOpenJDK 16

**Comment by Tkain** [comment]
[ForgeWrapper 1.5.0](https://github.com/ZekerZhayard/ForgeWrapper/releases/tag/1.5.0) fixes this issue. Right now it's just a matter of waiting until the new version makes its way to MultiMC's servers.

(If you're adventurous, you can patch it in yourself by clicking 'customize' on the Forge installation and editing the JSON as appropriate, but it'll probably only be a couple hours before it gets updated on their end.)

**Comment by vPubneutral** [comment]
> I think it's because they force the use of minecraft (official launcher) game profiles (at least that's what the error looks like to me)
> happens here aswell.
> 
> Also what's your Java runtime?
> Mine's AdoptOpenJDK 16

I use Oracle's JRE 8u301 for 1.16, but I did also try using AdoptOpenJDK 16 (forge supports it with some messing around with JVM arguments), but it had the same error.


> [ForgeWrapper 1.5.0](https://github.com/ZekerZhayard/ForgeWrapper/releases/tag/1.5.0) fixes this issue. Right now it's just a matter of waiting until the new version makes its way to MultiMC's servers.
> 
> (If you're adventurous, you can patch it in yourself by clicking 'customize' on the Forge installation and editing the JSON as appropriate, but it'll probably only be a couple hours before it gets updated on their end.)

I was able to get this to work by changing the normal url with a direct link to the .jar download you gave here, but I also had to manually get the sha1 hash and file size since it isn't on the release page like it was for 1.4.2. If anyone wants to try this for themselves, I'll post the correct info here:

Replace
```
{
            "downloads": {
                "artifact": {
                    "sha1": "79ff9c1530e8743450c5c3ebc6e07b535437aa6e",
                    "size": 22346,
                    "url": "https://files.multimc.org/maven/io/github/zekerzhayard/ForgeWrapper/1.4.2/ForgeWrapper-1.4.2.jar"
                }
            },
            "name": "io.github.zekerzhayard:ForgeWrapper:1.4.2"
        },
```

With
```
{
            "downloads": {
                "artifact": {
                    "sha1": "0E0721BF4A7B10F12329ACFEDFCB3A44A27E29F6",
                    "size": 27432,
                    "url": "https://github.com/ZekerZhayard/ForgeWrapper/releases/download/1.5.0/ForgeWrapper-1.5.0.jar"
                }
            },
            "name": "io.github.zekerzhayard:ForgeWrapper:1.5.0"
        },
```

Since it works and it'll probably get updated on it's own soon, I'm going to go ahead and close this.

**Comment by phit** [comment]
Gonna leave this open until it's actually fixed.

**Comment by Tkain** [comment]
Figured I should update the JSON edits for ForgeWrapper 1.5.1. 1.5.0 works just fine, but just in case you want the latest:

```
        {
            "downloads": {
                "artifact": {
                    "sha1": "0e0721bf4a7b10f12329acfedfcb3a44a27e29f6",
                    "size": 29892,
                    "url": "https://github.com/ZekerZhayard/ForgeWrapper/releases/download/1.5.1/ForgeWrapper-1.5.1.jar"
                }
            },
            "name": "io.github.zekerzhayard:ForgeWrapper:1.5.1"
        },
```

**Comment by vPubneutral** [comment]
Update: Newer Forge versions for 1.16.5 are listed in MultiMC, and they launch just fine. Playing with 36.2.2 right now!

![wowie](https://user-images.githubusercontent.com/19579178/127230604-87ccb890-3d4f-49fd-bf2d-e4339aa0f6f1.PNG)

No, Forge for 1.17.1 does _not_ work in MultiMC (yet.) I'm sure they're working on it. Since this issue is directly mentioned in another issue about Forge for 1.17.1 and some people may find there way here from there, I figured it couldn't hurt to say this.

**Comment by Tkain** [comment]
You might want to change the issue title to reflect the fact that that Forge 37 (1.17.1) is now the only version that can't be played.

**Comment by vPubneutral** [comment]
> You might want to either change the issue title to reflect the fact that that Forge 37 (1.17.1) is now the only version that can't be played or close this issue if this was meant to be 1.16-specific.

Yeah, this was meant to be 1.16 specific. Forge for 1.17.1 wasn't out when I opened this, but I'm guessing it got mentioned once we figured out that ForgeWrapper was the cause of not only my original issue but also (partly, at least) the reason 1.17 Forge doesn't launch in MultiMC. I meant to close it, just forgot. I'll close with comment.

> Also, if this _does_ cover Forge 37, is this redundant with #3936, or will more work need to be done aside from that?

I have absolutely no idea, but #3936 was opened by ZekerZhayard, the developer of ForgeWrapper, who definitely knows way more about any of this than I do, so I figure that if that issue is for 1.17 specifically, it makes more sense for 1.17-related discussion to happen there, not here.
